aws/resource_aws_iam_role_policy_attachment.go

```diff
@@ -16,6 +15,9 @@ func resourceAwsIamRolePolicyAttachment() *schema.Resource {
 		Create: resourceAwsIamRolePolicyAttachmentCreate,
 		Read:   resourceAwsIamRolePolicyAttachmentRead,
 		Delete: resourceAwsIamRolePolicyAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsIamRolePolicyAttachmentImport,
```

Comment:
Instead of implementing a "complex" import function (importing all attached policies), which can be dangerous, we should instead support importing the resource 1:1.

This can be implemented via (which expects a format ROLENAME_POLICYARN):

```go
		Importer: &schema.ResourceImporter{
			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
				idParts := strings.Split(d.Id(), "_")
				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
					return nil, fmt.Errorf("unexpected format of ID (%q), expected ROLENAME_POLICYARN", d.Id())
				}
				roleName := idParts[0]
				policyARN := idParts[1]
				d.Set("policy_arn", policyARN)
				d.Set("role_name", roleName)
				d.SetId(resource.PrefixedUniqueId(fmt.Sprintf("%s-", roleName)))
				return []*schema.ResourceData{d}, nil
			},
		},
```

And tested via:

```go
// new function
func testAccAWSIAMRolePolicyAttachmentImportStateIdFunc(resourceName string) resource.ImportStateIdFunc {
	return func(s *terraform.State) (string, error) {
		rs, ok := s.RootModule().Resources[resourceName]
		if !ok {
			return "", fmt.Errorf("not found: %s", resourceName)
		}

		return fmt.Sprintf("%s_%s", rs.Primary.Attributes["role_name"], rs.Primary.Attributes["policy_arn"]), nil
	}
}

// As a new TestStep in TestAccAWSRolePolicyAttachment_basic
{
	ResourceName:      "aws_iam_role_policy_attachment.test-attach",
	ImportState:       true,
	ImportStateIdFunc: testAccAWSIAMRolePolicyAttachmentImportStateIdFunc("aws_iam_role_policy_attachment.test-attach"),
	ImportStateVerify: true,
},
```